funding/batch.go

```diff
@@ -293,7 +302,26 @@ func (b *Batcher) BatchFund(ctx context.Context,
 	// channel funding negotiation with the peers. Because we specified a
 	// PSBT assembler, we'll get a special response in the channel once the
 	// funding output script is known (which we need to craft the TX).
-	eg := &errgroup.Group{}
+	// In case one or more of the peers fail to respond within a timeout
+	// period, we'll abort the whole process. During the timeout period, the
+	// user can abort the funding process manually by sending a Ctrl+C to
+	// the process.
+
+	// We use an errgroup to keep track of all goroutines that we launch.
+	eg, egCtx := errgroup.WithContext(ctx)
+
+	// We also launch a goroutine that waits for the errgroup context to be
+	// cancelled, through a Ctrl+C for example. If this happens during the
+	// DefaultBatchFundingTimeout period, we'll abort the whole funding
+	// process.
+	go func() {
```

Comment:
This is a bit weird of a pattern, even though it seems to be working.
I think we should just pass in the error group context as the parent context into `waitForUpdate` which achieves the same.
See this proposed diff:
[8367.diff.txt](https://github.com/lightningnetwork/lnd/files/13917784/8367.diff.txt)

Reply:
Thanks for this simplification, this is much better. Seems to be working. Will create an itest soon.